keps/sig-cluster-lifecycle/wgs/2141-instance-specific-componentconfig/README.md

```diff
@@ -0,0 +1,616 @@
+  multiple YAML documents, and merge objects paired by Kind between the two.
+  We can consider supporting this in future iterations.
+
+### Test Plan
+
+ComponentConfig APIs are typically tested similarly to API server APIs, for
+example by testing round-trip conversions. Some tests specifically exercise
+the ability to share a config object, such as the Dynamic Kubelet Config tests.
+
+Components that opt-in to instance-specific Kinds should extend their existing
+tests to include the instance-specific Kinds, except where those tests are
+explicitly testing the ability to share a config object across concurrent
+instances.
+
+### Graduation Criteria
+
+A beta-versioned ComponentConfig API is using the instance-specific file,
+with strategic merge patch approach.
+
+### Upgrade / Downgrade Strategy
+
+Since instance-specific config is currently only exposed on the command-line,
+and no existing ComponentConfig at beta or later maturity levels need to be
+retrofitted to remove instance-specific parameters, immutable upgrades are
+relatively simple: just deploy the new resources with the new configuration
+format (this is how we approached upgrades for ComponentConfig in general).
+Immutable downgrades follow the same approach: deploy the new resources with
+the old version and old config.
+
+In-place upgrades, should not be significantly affected, though as usual
+upgrades should take care to make the in-place changes to the config before
+starting the new Kubelet version. If in-place downgrades are desired, the config
+should be backed up before making changes, so that it can be restored later.
+Correctly implementing in-place upgrades has always been an exercise for the
+user, and this KEP does not change that.
+
+Since backwards-compatibility of the command-line is maintained, in the short
+term no action will be necessary to upgrade a component without updating its
+config, though command-line flags will eventually be removed commensurate with
+the overall ComponentConfig strategy. At that time (which will permit for
+the required deprecation period and fair warning), it will be necessary to move
+to the file-based approach.
+
+### Version Skew Strategy
+
+Any given version of a component should always refer to a config version that
+it understands. Fortunately, this is relatively easy to coordinate via either
+the Pod spec () or node startup scripts. Wholly new Pods are created on
+DaemonSet upgrades, and if necessary the new set can refer to wholly new
+ConfigMaps or contain an updated Pod template. Wholly new nodes are typically
+created on node upgrades, and the new VMs can be deployed with updated config
+metadata.
+
+To be clear, an old Kubelet can't use the new config format, and shouldn't be
+configured to use it. A new Kubelet can use either the new format, or continue
+to use the old format until the deprecation period is up (in this case, when
+all other flags are also deprecated).
+
```

Comment:
PRR sections for alpha should also be completed. The section is missing from the kep currently, see template: https://github.com/kubernetes/enhancements/tree/master/keps/NNNN-kep-template#production-readiness-review-questionnaire

Reply:
Just do the best you can based on this specific KEP.